src/Component/FieldTypes/EnumFieldType.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\FieldTypes;
+
+use Sylius\Component\Grid\DataExtractor\DataExtractorInterface;
+use Sylius\Component\Grid\Definition\Field;
+use Sylius\Component\Grid\Exception\LogicException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class EnumFieldType implements FieldTypeInterface
+{
+    public function __construct(
+        private DataExtractorInterface $dataExtractor,
+        private ?TranslatorInterface $translator = null,
+    ) {
+    }
+
+    public function render(Field $field, $data, array $options): string
+    {
+        $enum = $this->dataExtractor->get($field, $data);
+
+        if (!$enum instanceof \BackedEnum) {
+            return $enum->name;
+        }
+
+        if (!$enum instanceof TranslatableInterface) {
+            return (string) $enum->value;
```

Comment:
I think we'll need a try catch here.

Like this one implemented by Florian on Callable (you'll need adapt the code, of course).
```
try {
    $value = (string) $value;
} catch (\Throwable $e) {
    throw new UnexpectedValueException(\sprintf(
        'Callable field (name "%s") returned value could not be converted to string: "%s".',
        $field->getName(),
        $e->getMessage(),
    ));
}
```

Reply:
Can't we safely assume a cast will work as the Backenum::$value property only allows int or string

> A Backed Enum may be backed by types of int or string, and a given enumeration supports only a single type at a time (that is, no union of int|string). 
https://www.php.net/manual/en/language.enumerations.backed.php